src/coreclr/jit/llvm.cpp

```diff
@@ -1825,9 +1825,9 @@ void Llvm::buildStoreObj(GenTreeObj* storeOp)
 
     CORINFO_CLASS_HANDLE structClsHnd  = structLayout->GetClassHandle();
     StructDesc*          structDesc    = getStructDesc(structClsHnd);
-    bool targetNotHeap = ((storeOp->gtFlags & GTF_IND_TGT_NOT_HEAP) != 0) || storeOp->Addr()->OperIsLocalAddr();
+    bool targetNotHeap = blockOp->OperIsBlk() || ((blockOp->gtFlags & GTF_IND_TGT_NOT_HEAP) != 0) || blockOp->Addr()->OperIsLocalAddr();
```

Comment:
```suggestion
    bool targetNotHeap = ((blockOp->gtFlags & GTF_IND_TGT_NOT_HEAP) != 0) || blockOp->Addr()->OperIsLocalAddr();
```
(See also the write barrier discussion)

Reply:
We dont need `targetNotHeap`  if we have the other check do we?

Reply:
Yep.